### PR TITLE
New version: GoLang.Go version 1.22.4

### PR DESCRIPTION
**Pull request by coolplaylinbot**
###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-pkgs/pull/156535)

### CODE REVIEW

**Comment by stephengillie** [comment]
Automatic Validation ended with:

> Installation failed with exit code -1978334957
> 2024-06-04 11:43:57.627 [CLI ] ShellExecute installer failed: 1633
> MSI (s) (84:10) [11:43:57:492]: Product: Go Programming Language arm64 go1.22.4 -- Installation failed.
> This installation package is not supported by this processor type. Contact your product vendor.
> 2024-06-04 11:44:13.937 [CLI ] ShellExecute installer failed: 1633
> MSI (s) (FC:18) [11:44:13:571]: Product: Go Programming Language arm64 go1.22.4 -- Installation failed.

Automated error analysis suggests 1633 may mean System Not Supported. Does this manifest have `Architecture: x86`? It might need to be `x64`.

(Automated response - build 883.)

File: manifests/g/GoLang/Go/1.22.4/GoLang.Go.installer.yaml [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+AppsAndFeaturesEntries:
+- DisplayName: Go Programming Language arm64 go1.22.4
+  UpgradeCode: '{21ADE9A3-3FDD-4BA6-BEA6-C85ABADC9488}'
+Installers:
+- Architecture: x86
+  InstallerUrl: https://go.dev/dl/go1.22.4.windows-arm64.msi
+  InstallerSha256: 553CC6C460F4E3EB4FAD5B897C0BB22CD8BBEB20929F0E3EEB939420320292CE
+- Architecture: x64
+  InstallerUrl: https://go.dev/dl/go1.22.4.windows-arm64.msi
+  InstallerSha256: 553CC6C460F4E3EB4FAD5B897C0BB22CD8BBEB20929F0E3EEB939420320292CE
```
Comment (stephengillie) [inline manifests/g/GoLang/Go/1.22.4/GoLang.Go.installer.yaml:30]:
```suggestion
- Architecture: x86
  InstallerUrl: https://go.dev/dl/go1.22.4.windows-386.msi
  InstallerSha256: 5C6446E2EA80BC6A971D2B34446F16E6517E638B0FF8D3EA229228D1931790B0
- Architecture: x64
  InstallerUrl: https://go.dev/dl/go1.22.4.windows-amd64.msi
  InstallerSha256: 3C21105D7B584759B6E266383B777CAF6E87142D304A10B539DBC66AB482BB5F
```

Comment (stephengillie) [reply manifests/g/GoLang/Go/1.22.4/GoLang.Go.installer.yaml:30]: CC @CoolPlayLin

**Comment by CoolPlayLin** [comment]
Incorrect manifest @russellbanks

**Review by stephengillie — APPROVED** [review]

**Pull request merged**